tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TaskMapper.java

```diff
@@ -15,42 +18,54 @@
 import org.springframework.stereotype.Service;
 import org.springframework.util.Assert;
 
+import java.util.Comparator;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Service
 public class TaskMapper {
-    private final String defaultInstanceType;
-
     private final Integer defaultHddSize;
-
-    @Autowired
     private MessageHelper messageHelper;
+    private final CloudPipelineAPIClient cloudPipelineAPIClient;
 
     private static final String SEPARATOR = " ";
     private static final String INPUT_TYPE = "input";
     private static final String OUTPUT_TYPE = "output";
     private static final String DEFAULT_TYPE = "string";
+    private static final String IMAGE = "image";
+    private static final String EXECUTORS = "executors";
+    private static final String ZONES = "zones";
+    private static final String MiB = "MiB";
     private static final Integer FIRST = 0;
+    private static final Integer ONLY_ONE = 1;
+    private static final Double GiB_TO_GiB = 1.0;
+    private static final Double MiB_TO_GiB = 0.0009765625;
 
-    public TaskMapper(@Value("${cloud.pipeline.instanceType}") String instanceType,
-                      @Value("${cloud.pipeline.hddSize}") Integer hddSize) {
-        this.defaultInstanceType = instanceType;
+
+    @Autowired
+    public TaskMapper(@Value("${cloud.pipeline.hddSize}") Integer hddSize,
+                      CloudPipelineAPIClient cloudPipelineAPIClient, MessageHelper messageHelper) {
         this.defaultHddSize = hddSize;
+        this.cloudPipelineAPIClient = cloudPipelineAPIClient;
+        this.messageHelper = messageHelper;
     }
 
     public PipelineStart mapToPipelineStart(TesTask tesTask) {
         PipelineStart pipelineStart = new PipelineStart();
         Map<String, PipeConfValueVO> params = new HashMap<>();
         TesExecutor tesExecutor = getExecutorFromTesExecutorsList(tesTask.getExecutors());
+        Assert.notNull(tesExecutor.getImage(), messageHelper.getMessage(
+                MessageConstants.ERROR_PARAMETER_NULL_OR_EMPTY, IMAGE));
+        Tool pipelineTool = loadToolByTesImage(tesExecutor.getImage());
 
+        pipelineStart.setInstanceType(getProperInstanceType(tesTask, pipelineTool));
         pipelineStart.setCmdTemplate(String.join(SEPARATOR, tesExecutor.getCommand()));
         pipelineStart.setDockerImage(tesExecutor.getImage());
         pipelineStart.setExecutionEnvironment(ExecutionEnvironment.CLOUD_PLATFORM);
         pipelineStart.setHddSize(defaultHddSize);
```

Comment:
we need to use this default value only if user hasn't specified hddSize

Reply:
fixed, added mapping for "hddSize" parameter in CP with "disk_Gb" in TES